code/particle/effects/GenericShapeEffect.h

```diff
@@ -156,6 +165,26 @@ class GenericShapeEffect : public ParticleEffect {
 		if (internal::required_string_if_new("+Number:", nocreate)) {
 			m_particleNum = ::util::parseUniformRange<uint>();
 		}
+		if (!nocreate) {
+			m_particleChance = 1.0f;
+		}
+		if (optional_string("+Chance:")) {
+			float chance;
+			stuff_float(&chance);
+			CLAMP(chance, 0.0f, 1.0f);
+			if (chance <= 0.0f) {
+				Error(LOCATION,
+					"Particle %s tried to set +Chance: %f\nChances below 0 would result in no particles.",
+					m_name.c_str(), chance);
+			} else if (chance > 1.0f) {
+				Warning(LOCATION,
+					"Particle %s tried to set +Chance: %f\nChances above 1 are ignored, please use +Number: (min,max) "
+					"to spawn multiple particles.", m_name.c_str(), chance);
+				chance = 1.0f;
+			}
+			m_particleChance = chance;
+		}
+		m_particleRoll = ::util::UniformFloatRange(m_particleChance - 1.0f, m_particleChance);
```

Comment:
Er... so if `m_particleChance` is `0.5f`, you want to generate random values between `0.5f` and `0.5f`?

Reply:
If m_particleChance is `0.5f`, then this is `(0.5f-1.0f, 0.5f)`, or `-0.5f` to `0.5f`.

Reply:
Ah, right, for some reason I was reading it as `1.0f - m_particleChance`; now it makes sense.